### PR TITLE
Update sha1.h

### PR DESCRIPTION
**Pull request by RosZad**
Minor documentation improvements:
*Standardized file brief description.
*Separated return statements.
*Reordered tags within documentation blocks so that params and returns are last in block.
*Applied previous function documentation improvements to depracated functions.

### CODE REVIEW

File: include/mbedtls/sha1.h [hunk 1]
```diff
@@ -1,7 +1,10 @@
 /**
  * \file sha1.h
  *
- * \brief The SHA-1 cryptographic hash function.
+ * \brief This file contains SHA-1 definitions and functions.
+ *
+ * The Secure Hash Algorithm 1 (SHA-1) cryptographic hash function is defined in 
+ * <em>FIPS 180-4: Secure Hash Standard (SHS)</em>.
```
Comment (sbutcher-arm) [inline include/mbedtls/sha1.h]: Can we provide a link here, if we're referencing the standard?

File: include/mbedtls/sha1.h [hunk 2]
```diff
@@ -66,69 +69,68 @@ mbedtls_sha1_context;
...
  * \brief          This function clones the state of a SHA-1 context.
  *
- * \param dst      The destination context.
- * \param src      The context to clone.
- *
  * \warning        SHA-1 is considered a weak message digest and its use
  *                 constitutes a security risk. We recommend considering
  *                 stronger message digests instead.
  *
+ * \param dst      The destination context.
```
Comment (hanno-arm) [inline include/mbedtls/sha1.h]: Suggestion: `The SHA-1 context to clone to.` and `The SHA-1 context to clone from.`.

File: include/mbedtls/sha1.h [hunk 3]
```diff
@@ -66,69 +69,68 @@ mbedtls_sha1_context;
...
  *
- * \param ctx      The context to initialize.
- *
- * \return         \c 0 if successful
- *
  * \warning        SHA-1 is considered a weak message digest and its use
  *                 constitutes a security risk. We recommend considering
  *                 stronger message digests instead.
  *
+ * \param ctx      The context to initialize.
```
Comment (hanno-arm) [inline include/mbedtls/sha1.h]: Suggestion: `SHA-1 context` instead of `context`.

File: include/mbedtls/sha1.h [hunk 4]
```diff
@@ -174,65 +175,67 @@ int mbedtls_internal_sha1_process( mbedtls_sha1_context *ctx,
...
- * \param ctx      The SHA-1 context to be initialized.
+ * \brief          This function starts a SHA-1 checksum calculation.
  *
  * \warning        SHA-1 is considered a weak message digest and its use
  *                 constitutes a security risk. We recommend considering
  *                 stronger message digests instead.
  *
+ * \deprecated     Superseded by mbedtls_sha1_starts_ret() in 2.7.0.
+ *
+ * \param ctx      The context to initialize.
```
Comment (hanno-arm) [inline include/mbedtls/sha1.h]: Suggestion: `SHA-1 context` instead of `context`.

File: include/mbedtls/sha1.h (hunk 2 above)
Comment (RosZad) [reply include/mbedtls/sha1.h]: done

File: include/mbedtls/sha1.h (hunk 3 above)
Comment (RosZad) [reply include/mbedtls/sha1.h]: done

File: include/mbedtls/sha1.h (hunk 4 above)
Comment (RosZad) [reply include/mbedtls/sha1.h]: done

File: include/mbedtls/sha1.h (hunk 1 above)
Comment (RosZad) [reply include/mbedtls/sha1.h]: Usually documentation does not link to other documents that are not under its control, that is - a link may change, and you would not be aware, so your documentation would contain broken links. 
I can add links, if you wish, but you would need to assign someone to verify these before every release, which I think is redundant overhead. These standards are easy to find.

**Pull request merged**